jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/HumanTaskHandlerHelper.java

```diff
@@ -201,6 +217,43 @@ public static Deadlines setDeadlines(Map<String, Object> parameters, List<Organi
 		return notifications;
 	}
 
+	protected static int getDeadlineRepeatLimit(String deadlineStr) {
+		String[] repeatableParts = DateTimeUtils.parseISORepeatable(deadlineStr);
+		String repeatLimit = repeatableParts[0];
+		if (!repeatLimit.isEmpty()) {
+			return Integer.parseInt(repeatLimit);
+		} else {
+			return -1;
+		}
+	}
+
+	protected static Date getDeadlineDurationDate(String durationStr) {
+		// handles iso repetable and both period and date notation
+		try {
+
+			if (DateTimeUtils.isRepeatable(durationStr)) {
+				String[] repeatableParts = DateTimeUtils.parseISORepeatable(durationStr);
+				String tempTimeDelay = repeatableParts[1];
+				String tempDateTimeStr = repeatableParts[2];
+
+				if (DateTimeUtils.isPeriod((tempDateTimeStr))) {
+					return new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + Duration.parse(tempDateTimeStr).toMillis());
+				} else {
+					return new Date(DateTimeUtils.parseDateTime(tempTimeDelay)+ DateTimeUtils.parseDateAsDuration(tempDateTimeStr));
+				}
+			} else {
+				if (DateTimeUtils.isPeriod((durationStr))) {
+					return new Date(System.currentTimeMillis() + Duration.parse(durationStr).toMillis());
+				} else {
+					return new Date(System.currentTimeMillis() + DateTimeUtils.parseDateAsDuration(durationStr));
```

Comment:
Same can go here - we can directly convert date into instant, then into millis and then into date.

Reply:
agree but then we would also need to add the try/catch already provided by DateTimeUtils.parseDateAsDuration. i rather reuse the existing methods i think.

Reply:
OK, true.